src/python/zquantum/qcbm/target_thermal_states_test.py

```diff
@@ -0,0 +1,246 @@
+import unittest
+import numpy as np
+from zquantum.core.utils import dec2bin, convert_tuples_to_bitstrings
+from zquantum.core.bitstring_distribution import BitstringDistribution
+from zquantum.core.bitstring_distribution.distance_measures.mmd import compute_mmd
+from zquantum.qcbm.target_thermal_states import (
+    int2ising,
+    ising2int,
+    get_random_hamiltonian_parameters,
+    get_thermal_target_distribution_dict,
+    get_target_bitstring_distribution,
+    cumulate,
+    sample,
+    get_thermal_sampled_distribution,
+    get_sampled_bitstring_distribution,
+)
+
+
+SEED = 14943
+
+
+class TestThermalTarget(unittest.TestCase):
+    def test_int2ising(self):
+        # Given
+        number = 4
+        length = 5
+        expected_ising_bitstring = [-1, -1, 1, -1, -1]
+
+        # When
+        ising_bitstring = int2ising(number, length)
+
+        # Then
+        self.assertListEqual(ising_bitstring, expected_ising_bitstring)
+
+    def test_ising2int(self):
+        # Given
+        ising_bitstring = [-1, -1, 1, -1, -1]
+        expected_number = 4
+
+        # When
+        number = ising2int(ising_bitstring)
+
+        # Then
+        self.assertEqual(number, expected_number)
+
+    def test_ising2int2ising(self):
+        # Given
+        expected_number = 14
+        length = 5
+
+        # When
+        number = ising2int(int2ising(expected_number, length))
+
+        # Then
+        self.assertEqual(number, expected_number)
+
+    def test_get_random_hamiltonian_parameters(self):
```

Comment:
I'm going to refrain from commenting on the rest of the tests since I expect more changes to `src/python/zquantum/qcbm/target_thermal_state.py`